### PR TITLE
feat(telephony-guides): add guides tab for telephony

### PR DESCRIPTION
**Pull request by Steffy29**
Ref: UXCT-1

### CODE REVIEW

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                     data-ng-if="$ctrl.hasGuides()"
+                     data-ng-repeat="section in $ctrl.guides.sections track by section.label">
+                <header class="widget-presentation-header">
+                    <h2 class="widget-presentation-title"
+                        data-translate="{{ section.label }}">
+                    </h2>
+                </header>
+                <ul>
+                    <li data-ng-repeat="guide in section.guides track by guide.label">
+                        <a title="{{ guide.label | translate }} ({{:: 'common_open_new_tab' | translate }})"
```
Comment (marie-j) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]:
```suggestion
                         <a 
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                           target="_blank"
+                           rel="noopener"
+                           data-ng-if="guide.url[$ctrl.language]"
+                           data-ng-href="{{ guide.url[$ctrl.language] }}">
+                            <span data-translate="{{ guide.label | translate }}"></span>
+                            <i class="ovh-font ovh-font-newtab" aria-hidden="true"></i>
```
Comment (marie-j) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]:
```suggestion
                             <span class="oui-icon oui-icon-external_link" aria-hidden="true"></span>
                             <span class="sr-only" data-translate="common_open_new_tab" aria-hidden="true"></span>
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+                  '{title}',
+                  _.snakeCase(this.$translate.instant(guide.label)),
+                );
+                this.count += 1;
+              }
+            });
+          }
+        });
+      }
+    }
```
Comment (marie-j) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]: Why is this necessary ? I don't see any `${title}` to replace in given URLs ?

File: client/app/telecom/telephony/billingAccount/telecom-telephony-billing-account.html [hunk 4]
```diff
@@ -61,6 +61,11 @@
                         <span data-translate="telephony_group_tab_contacts_manage"></span>
                     </responsive-tab>
 
+                    <responsive-tab data-state="telecom.telephony.guides"
+                                    data-dropdown-title="{{ 'telephony_group_tab_guides' | translate }}">
```
Comment (FredericEspiau) [inline client/app/telecom/telephony/billingAccount/telecom-telephony-billing-account.html]:
```suggestion
                                     data-dropdown-title="{{:: 'telephony_group_tab_guides' | translate }}">
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                    </h2>
+                </header>
+                <ul>
+                    <li data-ng-repeat="guide in section.guides track by guide.label">
+                        <a title="{{ guide.label | translate }} ({{:: 'common_open_new_tab' | translate }})"
+                           target="_blank"
+                           rel="noopener"
+                           data-ng-if="guide.url[$ctrl.language]"
+                           data-ng-href="{{ guide.url[$ctrl.language] }}">
+                            <span data-translate="{{ guide.label | translate }}"></span>
```
Comment (FredericEspiau) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]:
```suggestion
                             <span data-translate="{{:: guide.label | translate }}"></span>
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+                 data-ng-if="!$ctrl.hasGuides()"
+                 data-translate="telephony_guides_empty">
+            </div>
+
+            <section class="widget-presentation"
+                     data-ng-if="$ctrl.hasGuides()"
+                     data-ng-repeat="section in $ctrl.guides.sections track by section.label">
+                <header class="widget-presentation-header">
+                    <h2 class="widget-presentation-title"
+                        data-translate="{{ section.label }}">
```
Comment (FredericEspiau) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]:
```suggestion
                         data-translate="{{:: section.label }}">
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html (hunk 2 above)
Comment (Steffy29) [reply client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]: Sorry, but it's not the same style between the two versions of code

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js (hunk 3 above)
Comment (Steffy29) [reply client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]: I renamed the method and removed ${title}

File: client/app/telecom/telephony/billingAccount/guides/translations/Messages_fr_FR.xml [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+    <translation id="telephony_guides_manage_simultaneous_calling">Gérer et utiliser les appels simultanés</translation>
+    <translation id="telephony_guides_manage_number_presentation">Configurer la présentation de son numéro</translation>
+    <translation id="telephony_guides_import_phonebook">Gérer un carnet de contacts sur une ligne SIP</translation>
+    <translation id="telephony_guides_intercom_mode">Gérer le mode intercom de votre ligne</translation>
+    <translation id="telephony_guides_update_sounds">Modifier les musiques et sonneries de votre ligne</translation>
+    <translation id="telephony_guides_restrict_sip_line">Restreindre sa ligne SIP OVH par IP</translation>
+
+    <translation id="telephony_guides_section_phone_troubleshooting">Diagnostic et dépannage</translation>
+    <translation id="telephony_guides_phone_troubleshooting_pap">Dépanner son téléphone Plug and Phone</translation>
+</translations>
```
Comment (JeremyDec) [inline client/app/telecom/telephony/billingAccount/guides/translations/Messages_fr_FR.xml]: Be careful of missing required new line
```suggestion
</translations>

```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+      if (_.has(this.guides, 'sections')) {
+        _.forEach(this.guides.sections, (section) => {
+          if (_.has(section, 'guides')) {
+            _.forEach(section.guides, (guide) => {
+              if (_.has(guide, ['url', this.language]) && _.isString(guide.label)) {
+                this.count += 1;
+              }
+            });
+          }
+        });
```
Comment (marie-j) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]:
```suggestion
         this.count = _.flatmap(this.guides.section, section => section.guides).filter(guide => _.has(guide, `url.${this.language}`) && _.isString(guide.label)).length
```

**Comment by ovh-cds** [comment]
CDS Report build-manager#1082.0 ✘  
* Stage 1
  * Build dependencies and inject Sentry ✔ 
* Stage 2
  * Build manager CA ✔ 
  * Build manager EU ✘  
  * Build manager US ✔ 
  * Tests ✘

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js (hunk 8 above)
Comment (marie-j) [reply client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]: @Steffy29 What I meant is that you can replace the code above with the `if` and `forEach` by this

Comment (Steffy29) [reply client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]: No, _.flatMap is not available for our version of lodash

Comment (marie-j) [reply client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]: You can therefore `map` and then `flat` the resulting array

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+                </header>
+                <ul>
+                    <li data-ng-repeat="guide in section.guides track by guide.label">
+                        <a 
+                           title="{{ guide.label | translate }} ({{:: 'common_open_new_tab' | translate }})"
+                           target="_blank"
+                           rel="noopener"
+                           data-ng-if="guide.url[$ctrl.language]"
+                           data-ng-href="{{ guide.url[$ctrl.language] }}">
+                            <span data-translate="{{:: guide.label | translate }}"></span>
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]: You are using both `translate` directive and filter. Is it normal?

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+
+    <div class="text-center" data-ng-if="$ctrl.loading.init">
+        <oui-spinner></oui-spinner>
+    </div>
+
+    <div class="row"
+         data-ng-if="!$ctrl.loading.init">
+        <div class="col-xs-12 col-md-6">
+
+            <oui-message type="info" data-ng-if="!$ctrl.hasGuides()">{{'telephony_guides_empty' | translate}}</oui-message>
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]:
```suggestion
             <oui-message data-type="info" data-ng-if="!$ctrl.hasGuides()">{{'telephony_guides_empty' | translate}}</oui-message>
```

Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]: Could you please turn:

```html
{{'telephony_guides_empty' | translate}}
```

into:
```html
<span data-translate="telephony_guides_empty"></span>
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html [hunk 11]
```diff
@@ -0,0 +1,43 @@
+<section class="telecom-telephony-guides">
+
+    <div class="text-center" data-ng-if="$ctrl.loading.init">
+        <oui-spinner></oui-spinner>
+    </div>
+
+    <div class="row"
+         data-ng-if="!$ctrl.loading.init">
+        <div class="col-xs-12 col-md-6">
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.html]:
```suggestion
         <div class="col-md-6">
```

File: client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js [hunk 12]
```diff
@@ -0,0 +1,40 @@
+angular
+  .module('managerApp')
+  .controller('TelecomTelephonyGuidesCtrl', class TelecomTelephonyGuidesCtrl {
+    constructor($translate, TucToastError, TELEPHONY_GUIDES) {
```
Comment (antleblanc) [inline client/app/telecom/telephony/billingAccount/guides/telecom-telephony-guides.controller.js]: Both `$translate` and `TucToastError` aren't used.

```suggestion
    constructor(TELEPHONY_GUIDES) {
```
